core/src/filed/backup.cc

```diff
@@ -1605,6 +1605,13 @@ bool EncodeAndSendAttributes(JobControlRecord* jcr,
   {
     std::unique_lock l(jcr->mutex_guard());
     jcr->JobFiles++;                   /* increment number of files sent */
+    if (jcr->JobFiles > MAXIMUM_ALLOWED_FILES_PER_JOB) {
+      Jmsg1(jcr, M_FATAL, 0,
+            "FileIndex overflow detected. Please split the fileset to backup "
+            "less files.\n");
+      return false;
+    }
+
```

Comment:
This does not work.  You cannot be bigger than the max.

The correct way to check for this is
```
if (jcr->JobFiles == MAXIMUM_ALLOWED_FILES_PER_JOB) { ... }
jcr->JobFiles += 1;
```
Currently `jcr->JobFiles` will overflow causing the check to always be false.